synapse/notifier.py

```diff
@@ -453,6 +455,44 @@ def _notify_pusher_pool(self, max_room_stream_token: RoomStreamToken) -> None:
         except Exception:
             logger.exception("Error pusher pool of event")
 
+    @overload
+    def on_new_event(
+        self,
+        stream_key: Literal[StreamKeyType.ROOM],
+        new_token: RoomStreamToken,
+        users: Optional[Collection[Union[str, UserID]]] = None,
+        rooms: Optional[StrCollection] = None,
+    ) -> None:
+        ...
```

Comment:
I'm surprised we didn't already need this overload! Does it help avoid any asserts or anything?

Reply:
It doesn't reduce assertions, as it only limits what is accepted as arguments (it doesn't change the return type). I added it to catch the case where we passed in an `int` when using `StreamKeyType.RECEIPT`.